### ISSUE TITLE
"Exclude parent" option in RayCast3D has no effect

### ISSUE THREAD

**Issue by knil79** [godot-jolt/godot-jolt#644]
"Exclude parent" option RayCast3D has no effect doesn't seem to actually exclude the parent, I had to move the raycast outside of the parent. Not outside the node tree, but outside the physical bounds of the parent in 3D space.

**Comment by mihe** [comment]
Almost certainly a duplicate of #273.

tl;dr: Your usage of "Exclude Parent" probably doesn't work in Godot Physics either, due to how your particular scene is set up, and your ray-cast has until now only ignored it because Godot Physics (unlike Jolt) doesn't collide with the back/inside face of convex shapes when using `hit_back_faces`, which is `true` by default and (unfortunately) isn't exposed on `RayCast3D` in Godot 4.1 (but is in Godot 4.2).

So your options are to either fix your particular scene to make "Exclude Parent" actually do something, which it will only do if its _immediate_ parent is a `RigidBody3D`, `StaticBody3D` or `CharacterBody3D`, or if that's not an option then you can do the ray-cast through code instead where you'll be able to disable `hit_back_faces`, as seen [here](https://github.com/godot-jolt/godot-jolt/issues/273#issuecomment-1630485231).

If all of that sounds weird/wrong, then please provide a minimal repro project and I can take a closer look.

**Comment by knil79** [comment]
You are very likely right!
My tree looks like this 
![image](https://github.com/godot-jolt/godot-jolt/assets/18348392/8d7242d5-839f-4304-9c54-2092486086f8)
So as you said it probably worked before I used godot-jolt since Godot Physics doesn't collide with the convex shapes as you said.

My work around was to just move the raycast slightly outside the rigidbody and then it worked, no big deal! :)
